mlflow/server/js/src/graphql/graphql-codegen.ts

```diff
@@ -0,0 +1,57 @@
+import type { CodegenConfig } from '@graphql-codegen/cli';
+
+const config: CodegenConfig = {
+  schema: [`../graphql/do_not_modify_autogenerated_schema.gql`],
+  documents: ['src/**/!(*.test).js*', 'src/**/!(*.test).ts*'],
+  config: {
+    avoidOptionals: {
+      // Do not mark fields in the response as optional
+      field: true,
+    },
+    namingConvention: {
+      // Use ALL_CAPS for enum values
+      enumValues: 'change-case-all#constantCase',
+    },
+    // Always include __typename fields on response types
+    nonOptionalTypename: true,
+    // Do not include "Query"/"Mutation" in the type name
+    omitOperationSuffix: true,
+    // Only include enums and operations to avoid introducing duplicate types
+    onlyOperationTypes: true,
+    // Explicitly list out mapping for custom scalars from GQL type to TS type
+    scalars: {
+      GraphQLAnyValue: 'GraphQLGraphQLAnyValue',
+      Long: 'GraphQLLong',
+      LongString: 'GraphQLLongString',
+      StringMap: 'GraphQLStringMap',
+      WellKnownFieldMask: 'any',
+      WellKnownTimestamp: 'any',
+      GenaiDate: 'GraphQLGenaiDate',
```

Comment:
I wonder if we shouldn't ignore all `Genai*` types, they seem to be specific to mosaic schema and AFAICT they're used exclusively in queries related to finetuning, not being present in OSS

Reply:
Yeah let me take them out here, I only took them out in graphql.d.ts, not here